harness/temporalHelpers.js

```diff
@@ -1105,6 +1105,8 @@ var TemporalHelpers = {
         "11-18[U-CA=iso8601]",
         "11-18[u-CA=iso8601]",
         "11-18[FOO=bar]",
+        "+999999-01-01",
+        "-999999-01-01",
```

Comment:
These I'm not sure are correctly marked as invalid. By my reading of [ToTemporalMonthDay](https://tc39.es/proposal-temporal/#sec-temporal-totemporalmonthday):

- Step 11 creates an ISO Date Record { [[Year]]: 999999, [[Month]]: 1, [[Day]]: 1 }
- ISODateToFields in step 12 ignores the year because _type_ is MONTH-DAY, so we get a Calendar Date Record of { [[MonthCode]]: **"M01"**, [[Day]]: 1 } with all other fields UNSET
- CalendarMonthDayFromFields in step 14 passes the ISODateWithinLimits check, because it uses the 1972 reference year, having already discarded the 999999 year.

Reply:
The SM implementation has an [explicit check](https://searchfox.org/mozilla-central/rev/5ad94327fc03d0e5bc6aa81c0d7d113c2dccf947/js/src/builtin/temporal/PlainMonthDay.cpp#295-299) to reject too large dates, because for non-ISO 8601 calendars it's difficult to support dates too far into the past or future. I think there are two issues:
1. This explicit check should only apply for non-ISO 8601 calendar.
2. The spec itself should add a similar restriction. I'll prepare a spec PR.

Reply:
Sounds good. Would you mind splitting this change into a separate PR then? That'll likely take us longer to discuss and would need to be presented in TC39, but in the meantime we can merge the rest of this PR.